modules/objdetect/test/test_charucodetection.cpp

```diff
@@ -814,7 +814,7 @@ TEST(CharucoBoardGenerate, issue_24806)
         Point2f chessCorner(pixInSquare*(p.x/squareLength),
                             pixInSquare*(p.y/squareLength));
         Mat winCorner = chessboardZoneImg(Rect(Point(cvRound(chessCorner.x) - 1, cvRound(chessCorner.y) - 1), Size(2, 2)));
-        bool eq = (cv::countNonZero(goldCorner1 != winCorner) == 0) | (cv::countNonZero(goldCorner2 != winCorner) == 0);
+        bool eq = (cv::countNonZero(goldCorner1 != winCorner) == 0) || (cv::countNonZero(goldCorner2 != winCorner) == 0);
```

Comment:
BTW, `norm(a, b, NORM_INF) == 0` is faster and accurate than `countNonZero(a != b)`